autoendpoint/src/routers/fcm/router.rs

```diff
@@ -175,24 +175,30 @@ impl Router for FcmRouter {
             .get(&app_id)
             .ok_or_else(|| FcmError::InvalidAppId(app_id.clone()))?;
 
-        // GCM is the older message format for android, and it's not possible to generate
-        // new test keys. As of 2023-09, GCM should use FCM OAuth2 in order to authenticate.
-        // FCM operates by using the more complex token as part of an OAuth2
-        // transaction. According to the documentation, GCM and FCM are interoperable,
-        // meaning that the client provided tokens should match up, and that the structure
-        // of the data should also not make a huge difference.
-        //
-        // In Oct of 2023, "legacy" GCM support was deprecated. After some
-        // research, it was hypothesised that the current FCM project may be able
-        // to accept GCM routing identifiers as FCM message tokens. The current code presumes
-        // that this is a fragile relationship and so GCM errors do not immediately drop the
-        // remote user's endpoint (potentially causing a UA reset.)
-        //
-        let platform = if client.is_gcm {
-            "gcm_as_fcmv1"
-        } else {
-            "fcmv1"
-        };
+        if client.is_gcm {
```

Comment:
We don't have to rely on settings (which can just go away eventually) so the check can move to much earlier in e.g. this method.

In fact this check could alternatively go as early as inside of `extractors::validate_user` -- rejecting it there would skip potential expensive JWT validation entirely and a few other emitted metrics. We lack `app_id` over there for a verbose metric such as the one emitted by `handle_error` but I don't think we care much as it's all being dropped.

```suggestion
        let router_type = RouterType::from_str(&notification.subscription.user.router_type).unwrap_or(RouterType::FCM);
        if matches!(router_type, RouterType::GCM) {
```

Reply:
... thinking about this a bit more. I think we want to drop the subscription, not the user. It may be possible that the user has other valid subscriptions that are not GCM, but has one legacy, hold out endpoint that is. 

Dropping the user (in validate user) is a bit drastic.

Reply:
`handle_user` called w/ `RouterError::NotFound` as is done here is already dropping the user.

`router_type` is on the `User` in the router table: so UAIDs should be tied to one type of router, i.e. all of a user's channels should be of the same type?

Reply:
Yeah, I realized that after I made the initial comment, and why I went ahead and made the change. Plus, it's a moot point since the UA would effectively do the same thing once it did it's rectification.